modules/auth/proc-enabling-authentication-with-rhsso.adoc

```diff
@@ -0,0 +1,72 @@
+
+.. To register your {product-short} in RHSSO, in this realm, link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#registering-app_[create a Client ID], with:
+... **Client ID**: A distinctive client ID, such as __<{product-very-short}>__.
+... **Valid redirect URIs**: Set to the OIDC handler URL: `https://__<RHDH_URL>__/api/auth/oidc/handler/frame`.
+... Navigate to the **Credentials** tab and copy the **Client secret**.
+... Save the values for the next step:
+* **Client ID**
+* **Client Secret**
+
+.. To prepare for the verification steps, in the same realm, get the credential information for an existing user or link:https://docs.redhat.com/en/documentation/red_hat_single_sign-on/7.6/html-single/getting_started_guide/index#create-user_[create a user]. Save the user credential information for the verification steps.
+
+. To add your RHSSO credentials in your Developer Hub secrets, edit your Developer Hub secrets, such as `secrets-rhdh`, and add the following key/value pairs:
++
+`AUTH_OIDC_CLIENT_ID`:: Enter the saved **Client ID**.
+`AUTH_OIDC_CLIENT_SECRET`:: Enter the saved **Client Secret**.
+`AUTH_OIDC_METADATA_URL`:: Enter the saved **RHSSO realm base URL**.
+
+. To set up the RHSSO authentication provider in your {product-short} custom configuration, edit your custom Developer Hub config map, such as `app-config-rhdh`, and add following lines to the `app-config-rhdh.yaml` content:
++
+.`app-config-rhdh.yaml` fragment
+[source,yaml]
+----
+auth:
+  environment: production # <1>
+  providers:
+    oidc:
+      production:
+        metadataUrl: ${AUTH_OIDC_METADATA_URL} # <2>
+        clientId: ${AUTH_OIDC_CLIENT_ID} # <2>
+        clientSecret: ${AUTH_OIDC_CLIENT_SECRET} # <2>
+        ## Uncomment for additional configuration options
+        # callbackUrl: ${AUTH_OIDC_CALLBACK_URL}
+        # tokenEndpointAuthMethod: ${AUTH_OIDC_TOKEN_ENDPOINT_METHOD}
+        # tokenSignedResponseAlg: ${AUTH_OIDC_SIGNED_RESPONSE_ALG}
+        # scope: ${AUTH_OIDC_SCOPE}
+        ## Declarative resolvers to override the default resolver: `emailLocalPartMatchingUserEntityName`
+        ## The authentication provider tries each sign-in resolver until it succeeds, and fails if none succeed. Uncomment the resolvers that you want to use.
+        # signIn:
+        #   resolvers:
+        #     - resolver: preferredUsernameMatchingUserEntityName
+        #     - resolver: emailMatchingUserEntityProfileEmail
+        #     - resolver: emailLocalPartMatchingUserEntityName
+signInPage: oidc # <3>
+dangerouslyAllowSignInWithoutUserInCatalog: true # <4>
+----
++
+<1> Mark the environment as `production` to hide the Guest login in the {product-short} home page.
+<2> To configure the OIDC provider with your secrets.
+<3> To enable the OIDC provider as default sign-in provider.
+<4> To enable authentication without requiring to provision the {product-short} software catalog.
```

Comment:
```suggestion
<4> To enable authentication without requiring to provision users in the {product-short} software catalog.
```
This example goes though a non-prod set up.  Is there a follow up issue to show how to ingest users for a prod set up?  This would require refactoring the [instructions](https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.2/html/configuring_plugins_in_red_hat_developer_hub/rhdh-installing-dynamic-plugins#rhdh-keycloak_plugin-rhdh) to set up keycloak plugin